### PR TITLE
[TASK] Switch eventlisteners  to constructor property promotion

### PR DESCRIPTION
**Pull request by linawolf**
And remove redundant readonly declarations

Releases: main, 12.4

### CODE REVIEW

File: Documentation/ApiOverview/Events/Events/Backend/_ModifyLinkExplanationEvent/_MyEventListener.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 final readonly class MyEventListener
 {
     public function __construct(
-        private readonly IconFactory $iconFactory,
+        private IconFactory $iconFactory,
```
Comment (froemken) [inline Documentation/ApiOverview/Events/Events/Backend/_ModifyLinkExplanationEvent/_MyEventListener.php:18]: Ahh...there is a static $iconCache in IconFactory. Good catch

File: Documentation/ApiOverview/Events/Events/Backend/_ModifyRecordListTableActionsEvent/_MyEventListener.php [hunk 2]
```diff
@@ -24,12 +24,9 @@
...
 final readonly class MyEventListener
 {
-    private LoggerInterface $logger;
-
-    public function __construct(LoggerInterface $logger)
-    {
-        $this->logger = $logger;
-    }
+    public function __construct(
+        private LoggerInterface $logger,
```
Comment (froemken) [inline Documentation/ApiOverview/Events/Events/Backend/_ModifyRecordListTableActionsEvent/_MyEventListener.php:28]: Just search in core. They also use the "readonly" keyword

```suggestion
        private readonly LoggerInterface $logger,
```

File: Documentation/ApiOverview/Events/Events/Backend/_ModifyLinkExplanationEvent/_MyEventListener.php (hunk 1 above)
Comment (froemken) [inline Documentation/ApiOverview/Events/Events/Backend/_ModifyLinkExplanationEvent/_MyEventListener.php:18]: Just searched in core. They also use the "readonly" keyword

```suggestion
        private readonly IconFactory $iconFactory,
```

**Review by froemken — COMMENTED** [review]
I would leave "readonly"

**Comment by linawolf** [comment]
If the class is readonly the readonly in the constructor is superfluous?

**Comment by froemken** [comment]
Wow...good point. Never thought about that.

**Comment by froemken** [comment]
I have asked Gemini and ChatCGT. Both tell me it will work without "readonly", too. I have marked my comments as solved.

**Review by froemken — APPROVED** [review]

**Pull request merged**